src/main/java/org/apache/maven/shared/incremental/IncrementalBuildHelper.java

```diff
@@ -323,25 +362,23 @@ public String[] beforeRebuildExecution( IncrementalBuildHelperRequest incrementa
     public void afterRebuildExecution( IncrementalBuildHelperRequest incrementalBuildHelperRequest )
         throws MojoExecutionException
     {
-        DirectoryScanner diffScanner = getDirectoryScanner();
-        // now scan the same directory again and create a diff
-        diffScanner.scan();
-        DirectoryScanResult scanResult = diffScanner.diffIncludedFiles( filesBeforeAction );
-
         File mojoConfigBase = getMojoStatusDirectory();
-        File mojoConfigFile = new File( mojoConfigBase, CREATED_FILES_LST_FILENAME );
 
-        try
-        {
-            FileUtils.fileWriteArray( mojoConfigFile, scanResult.getFilesAdded() );
-        }
-        catch ( IOException e )
-        {
-            throw new MojoExecutionException( "Error while storing the mojo status", e );
-        }
+        writeChangedFiles(
+                getDirectoryScanner(),
```

Comment:
These getter methods don't seem to help. They're an unnecessary layer of indirection. Just use the constructor directly.

Reply:
So yeah I was just going for consistency with the existing API!

Reply:
That only makes sense if the existing API is correct. Consistency with bad code is not a goal.